extensions-core/hdfs-storage/src/main/java/io/druid/storage/hdfs/HdfsDataSegmentKiller.java

```diff
@@ -45,7 +45,7 @@
   public HdfsDataSegmentKiller(final Configuration config, final HdfsDataSegmentPusherConfig pusherConfig)
   {
     this.config = config;
-    this.storageDirectory = new Path(pusherConfig.getStorageDirectory());
+    this.storageDirectory = new Path(HdfsDataSegmentUtil.getFullyQualifiedHdfsPath(pusherConfig.getStorageDirectory(), config));
```

Comment:
May be I'm missing something, can you please explain 
why this shouldn't be needed?